File: hangman_py/GuessWord.py
from Guess import Guess

# A Guess that represents guessing a word for the current Hangman game

class GuessWord(Guess):
  def __init__(self, guess):
    self.guess = guess

  def makeGuess(self, game):
    game.guessWord(self.guess)

  def __str__(self):
    return "GuessWord[%s]" % self.guess

File: hangman_lua/GuessWord.lua
module(..., package.seeall)

Guess = require "Guess"
cl = require "cl"

-- A Guess that represents guessing a word for the current Hangman game

GuessWord = cl.makeClass({
  init = function(self, guess)
    self.guess = guess
  end,

  makeGuess = function(self, game)
    game:guessWord(self.guess)
  end,

  __tostring = function(self)
    return string.format("GuessWord[%s]", self.guess)
  end

}, Guess)

return GuessWord
